Sources/Tentacle/Label.swift

```diff
@@ -7,17 +7,31 @@
 //
 
 import Foundation
-import Argo
-import Curry
-import Runes
 
-public struct Label: CustomStringConvertible {
+public struct Label: CustomStringConvertible, ResourceType {
     public let name: String
     public let color: Color
 
     public var description: String {
         return name
     }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.name = try container.decode(String.self, forKey: .name)
+        self.color = Color(hex: try container.decode(String.self, forKey: .color))
+    }
+
+    public init(name: String, color: Color) {
+        self.name = name
+        self.color = color
+    }
+
+    // TODO: remove this, or replace `Color` with something (De)codable
+    private enum CodingKeys: String, CodingKey {
```

Comment:
I can't reference CodingKeys in the `init` unless I make a custom `Color` type that's Decodable (instead of just being an alias to `UI/NSColor`. I'll add that in a minute.

Reply:
Thinking about this further, I think we should be explicit with `CodingKeys` to avoid any breakage in future refactoring. If Matt has different ideas we can always remove them in a future PR.

Reply:
I'm fine with being explicit. 👍